### DISCUSSION TITLE
Script with beforeInteractive strategy makes Cypress to throw Hydration error

### DISCUSSION THREAD

**Discussion by abaracedo** [vercel/next.js#55729] [Help] [answered]
### Summary

I noticed that when a Script component is using `strategy="beforeInteractive"` Cypress is throwig the error: `Hydration failed because the initial UI does not match what was rendered on the server.`.

I know the error comes from using this strategy because omitting it Cypress works. I can't omit this strategy as I need to load an script before anything else loads; I tried to set `suppressHydrationWarning` in the Script component but it doesn't work...

I'd like to know if anybody had this issue and found a solution to make it work.

### Additional information

```js
I also found in Sentry the same Hydration error that happened to some users.

I'm using next v13.4.8
```

**Comment by avsmithy** [comment]
I'm facing the same issue. Cypress is logging hydration warnings and also hanging occasionally when using beforeInteractive. @abaracedo did you find a solution/workaround?

**Reply by abaracedo** [reply, marked as answer]
I found the issue was caused by third-party scripts (Google Tag Manager, VWO, etc...). 

As I don't need them to run cypress, I create a `CYPRESS_IGNORE_THIRD_PARTY_SCRIPTS` env var when I run the E2E pipeline. In the `<head>` tag I added a condition for not including these scripts if I'm running Cypress.

```jsx
{!process.env.CYPRESS_IGNORE_THIRD_PARTY_SCRIPTS && (
  <>
    {/* third party scripts */}
  </>
)
```

**Reply by avsmithy** [reply]
Understood, thank you for the advice.
